### DISCUSSION TITLE
Import-Excel - directly from Base64

### DISCUSSION THREAD

**Discussion by sebastijanz** [dfinke/ImportExcel#1370] [Q&A]
Hello, would be possible to use Import-Excel directly from Base64 format? I'm parsing excel files from exchange 365 and then saving them locally.  
When they are saved I'm importing them using "Import-Excel" and saving data to database.

Now I have an idea to skip saving files locally. :)

**Comment by dfinke** [comment]
Unfortunately, the .NET C# library I use does not support that or `MemoryStream` structures.  If it did, it could do that directly.

Off the top of my head, could you open the file, convert the base 64 and save it as an xlsx?

**Reply by sebastijanz** [reply]
Yes, Im already saving it and then importing it via "Import-Excel". I thought maybe I could "bypass" saving file. :)

**Reply by dfinke** [reply]
yeah, if EPPlus could handle memory streams that scenario could be solved and a few other cool ones.
